{
    "comments": [
        {
            "author": null,
            "body": "This has most likely been fixed in 5.5.10.",
            "date": "20050922T19:26:31",
            "id": 0
        }
    ],
    "component": "Catalina",
    "description": "This happens in 5.5.9 and 5.5.8 on solaris and fedora linux but works fine in\n5.0.28. Creating symlink under app's root directory causes an exception when\ntrying to view directory. For instance I can do:\n\ncd webapps/jsp-examples/colors\nmkdir temp\nln -s temp temp1\n\nand when I http://localhost:8080/jsp-examples/colors\n\nit produces this:\n\n2005-08-18 14:15:01,430 [http-8080-Processor25] ERROR\norg.apache.catalina.core.ContainerBase.[Catalina].[local\nhost].[/apres].[default] - Servlet.service() for servlet default threw exception\njava.lang.NullPointerException\n        at\norg.apache.catalina.servlets.DefaultServlet.renderHtml(DefaultServlet.java:1345)\n        at\norg.apache.catalina.servlets.DefaultServlet.render(DefaultServlet.java:1102)\n        at\norg.apache.catalina.servlets.DefaultServlet.serveResource(DefaultServlet.java:807)\n        at\norg.apache.catalina.servlets.DefaultServlet.doGet(DefaultServlet.java:335)\n        at javax.servlet.http.HttpServlet.service(HttpServlet.java:689)\n        at javax.servlet.http.HttpServlet.service(HttpServlet.java:802)\n        at\norg.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:252)\n        at\norg.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:173)\n        at\norg.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:213)\n        at\norg.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:178)\n        at\norg.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:126)\n        at\norg.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:105)\n        at\norg.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:107)\n        at\norg.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:148)\n        at\norg.apache.coyote.http11.Http11Processor.process(Http11Processor.java:856)\n        at\norg.apache.coyote.http11.Http11Protocol$Http11ConnectionHandler.processConnection(Http11Protocol.ja\nva:744)\n        at\norg.apache.tomcat.util.net.PoolTcpEndpoint.processSocket(PoolTcpEndpoint.java:527)\n        at\norg.apache.tomcat.util.net.LeaderFollowerWorkerThread.runIt(LeaderFollowerWorkerThread.java:80)\n        at\norg.apache.tomcat.util.threads.ThreadPool$ControlRunnable.run(ThreadPool.java:684)\n        at java.lang.Thread.run(Thread.java:595)",
    "hasPatch": false,
    "hasScreenshot": false,
    "id": "36254",
    "issuetypeClassified": "BUG",
    "issuetypeTracker": "BUG",
    "priority": "P2 normal",
    "product": "Tomcat 5",
    "project": "TOMCAT",
    "summary": "Creating symlink causes NullPointerException",
    "systemSpecification": true,
    "version": "5.5.9"
}